### PR TITLE
Discv5: Do not issue FINDNODES right after the start

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Sometimes we're not able to send/receive discv5 packets successfully right after it starts. Ideally we should investigate discv5 for that, but I found we all `await` for the discv5 start process correctly.

```
2021-11-11T11:44:03.605Z discv5:service Starting discv5 service with node id a94979c6e3449280c849ddfcde174ddadd3cd4db280193d28743aeeef291a3fb
2021-11-11T11:44:03.605Z discv5:sessionService Starting session service with node id a94979c6e3449280c849ddfcde174ddadd3cd4db280193d28743aeeef291a3fb
2021-11-11T11:44:03.611Z discv5:service Starting a new lookup. Id: 1
```

**Description**

+ Skip doing `FINDNODES` in the 1st heart beat of peer manager

part of #3423

### CODE REVIEW

File: packages/lodestar/src/network/peers/peerManager.ts [hunk 1]
```diff
@@ -113,6 +113,8 @@ export class PeerManager {
 
   private seenPeers = new Set<string>();
 
+  private firstHearbeat = true;
```
Comment (dapplion) [inline packages/lodestar/src/network/peers/peerManager.ts]: We could generalize this into N seconds after starting discv5, and fine tune accordingly

File: packages/lodestar/test/e2e/network/peers/peerManager.test.ts [hunk 2]
```diff
@@ -85,7 +85,12 @@ describe("network / peers / PeerManager", function () {
         attnetsService: mockSubnetsService,
         syncnetsService: mockSubnetsService,
       },
-      {targetPeers: 30, maxPeers: 50, discv5: null}
+      {
+        targetPeers: 30,
+        maxPeers: 50,
+        discv5: null,
+        firstHeartBeatDelayMs: 1,
```
Comment (dapplion) [inline packages/lodestar/test/e2e/network/peers/peerManager.test.ts:92]: Miliseconds, should be 1000 not 1

File: packages/lodestar/test/e2e/network/network.test.ts [hunk 3]
```diff
@@ -55,6 +55,7 @@ describe("network", function () {
       targetPeers: 1,
       bootMultiaddrs: [],
       localMultiaddrs: [],
+      firstHeartBeatDelayMs: 1,
```
Comment (dapplion) [inline packages/lodestar/test/e2e/network/network.test.ts]: Make it 0 for clarity, on all the tests

File: packages/lodestar/src/network/peers/peerManager.ts [hunk 4]
```diff
@@ -45,6 +46,11 @@ export type PeerManagerOpts = {
   targetPeers: number;
   /** The maximum number of peers we allow (exceptions for subnet peers) */
   maxPeers: number;
+  /**
+   * Delay the 1st heartbeat after starting discv5
+   * See https://github.com/ChainSafe/lodestar/issues/3423
+   */
+  firstHeartBeatDelayMs: number;
```
Comment (dapplion) [inline packages/lodestar/src/network/peers/peerManager.ts]: Do we want to delay the whole heartbeat or only discv5 queries? The hearbeat is doing other important actions like disconnecting bad peers that is unrelated to this issue.

Comment (dapplion) [reply packages/lodestar/src/network/peers/peerManager.ts]: Another approach is to record the start time of discv5 and in this block:

https://github.com/ChainSafe/lodestar/blob/c5cef290fd3eab076fc9872bce05d10f913e7dc0/packages/lodestar/src/network/peers/peerManager.ts#L403-L405

Check

```ts
if (Date.now() - this.discv5StartMs > this.discv5FirstQueryDelayMs) {
  // do query
}
```

Comment (dapplion) [reply packages/lodestar/src/network/peers/peerManager.ts]: You can even move this code into the `PeerDiscovery` class to be closer to discv5 logic

Comment (twoeths) [reply packages/lodestar/src/network/peers/peerManager.ts]: discv5 queries is part of the `heartbeat` so I delayed the whole heartbeat only at the 1st time to make it simple. The default delay in ms is 1000 (1s)

**Comment by twoeths** [comment]
@dapplion even through we define `discv5FirstQueryDelayMs`, it'll always start issueing `FINDNODES` request from the 2nd heartbeat
```
021-11-15T10:10:51.530Z discv5:service Starting discv5 service with node id 5ba50119c3f282470441414915f897f587c1fb48ac2bdaef82039a9dc8053038
2021-11-15T10:10:51.531Z discv5:sessionService Starting session service with node id 5ba50119c3f282470441414915f897f587c1fb48ac2bdaef82039a9dc8053038


2021-11-15T10:11:21.539Z discv5:service Starting a new lookup. Id: 1
```

File: packages/lodestar/src/network/peers/discover.ts [hunk 5]
```diff
@@ -196,6 +202,10 @@ export class PeerDiscovery {
    * Request to find peers. First, looked at cached peers in peerStore
    */
   private async runFindRandomNodeQuery(): Promise<void> {
+    // Delay the 1st query after starting discv5
+    if (Date.now() - this.discv5StartMs <= this.discv5FirstQueryDelayMs) {
+      return;
+    }
```
Comment (dapplion) [inline packages/lodestar/src/network/peers/discover.ts:209]: Whitespace after this if please :pray:

File: packages/lodestar/src/network/peers/discover.ts [hunk 6]
```diff
@@ -196,6 +202,10 @@ export class PeerDiscovery {
    * Request to find peers. First, looked at cached peers in peerStore
    */
   private async runFindRandomNodeQuery(): Promise<void> {
+    // Delay the 1st query after starting discv5
```
Comment (dapplion) [inline packages/lodestar/src/network/peers/discover.ts:205]: Link to the Github issue that explain why this is necessary

**Comment by dapplion** [comment]
> even through we define `discv5FirstQueryDelayMs`, it'll always start issueing `FINDNODES` request from the 2nd heartbeat

Makes sense, not a problem :+1: but now we can add longer delays if we need that would go beyond the second heartbeat

**Review by dapplion — APPROVED** [review]
